puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/nightly_packaging.groovy

```diff
@@ -0,0 +1,30 @@
+def setup_nightly_build_environment(args) {
+    def commit_hash = ''
+
+    dir(args.package_name) {
+        git(url: "https://github.com/${args.github_repo}.git", branch: 'develop')
+        commit_hash = archive_git_hash()
+    }
+    dir('foreman-packaging') { git(url: 'https://github.com/theforeman/foreman-packaging.git', branch: 'rpm/develop') }
+    setup_obal()
+    configureRVM(args.ruby_version ?: env.ruby_version)
+
+    return commit_hash
+}
+
+def generate_nightly_sourcefiles(args) {
+    def sourcefile_paths = []
+
+    dir(args.package_name) {
+        withRVM(["bundle install --jobs 5 --retry 5"], env.ruby_version)
+        withRVM(["bundle exec rake pkg:generate_source"], env.ruby_version)
+        archiveArtifacts(artifacts: 'pkg/*')
+        dir('pkg') {
+            sourcefile_paths = sh(script: "ls -1", returnStdout: true).trim().tokenize('\n').collect {
```

Comment:
we have similar code in
```
puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/foreman_infra.groovy:    def files_list = sh(script: "ls jobs/ -1", returnStdout: true).trim().split()
puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/koji.groovy:        tasks = sh(returnStdout: true, script: "ls kojilogs -1 |grep -o '[0-9]*\$'").trim().split()
```
should we make this more generic and a library function?

Reply:
Good idea